Sources/UIScrollView.swift

```diff
@@ -77,6 +71,50 @@ open class UIScrollView: UIView {
             addSubview($0)
         }
     }
+    
+    // When adding subviews, make sure that the scroll indicators stay on top.
+    // Subview list goes back-to-front, so indicators need to take the last two positions.
+    // We use the earliestIndicatorInSubviewHierarchy variable to make sure new views are always inserted below them.
+    private lazy var earliestIndicatorInSubviewHierarchy: UIView? = {
+        let indexOfHorizontal = subviews.index(of: horizontalScrollIndicator)
+        let indexOfVertical = subviews.index(of: verticalScrollIndicator)
+
+        // if both indicators are missing, return nil, if one is missing, return one that's present
+        if indexOfVertical == nil && indexOfHorizontal == nil { return nil }
+        if indexOfHorizontal == nil { return verticalScrollIndicator }
+        if indexOfVertical == nil { return horizontalScrollIndicator }
+
+        // if both indicators are in the view hierarchy, return the one that is last
+        if indexOfHorizontal! < indexOfVertical! {
+            return horizontalScrollIndicator
+        } else {
+            return verticalScrollIndicator
+        }
+    }()
+
+
+    open override func addSubview(_ view: UIView) {
+        // If the view we're adding is an indicator, just add it.
+        // If not, check for indicators and insert below the last one present.
+        if view == horizontalScrollIndicator || view == verticalScrollIndicator {
```

Comment:
this can be simplified too:

```
if view !== horizontalScrollIndicator, view !== verticalScrollIndicator, let earliestIndicatorInSubviewHierarchy = earliestIndicatorInSubviewHierarchy  {
    super.insertSubview(view, belowSubview: earliestIndicatorInSubviewHierarchy)
} else {
    super.addSubview(view)
}
```

Reply:
done